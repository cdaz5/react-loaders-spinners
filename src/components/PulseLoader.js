import styled, { keyframes } from 'styled-components';
import { string, number, bool } from 'prop-types';

const pulse = keyframes`
  0% {
    transform: scale(0);
    opacity: 1;
  }
  100% {
    transform: scale(1.3);
    opacity: 0;
  }
`;

const PulseLoader = styled.div`
	display: ${props => (props.loading ? 'flex' : 'none')};
	width: ${props => (props.width ? `${props.width}px` : '100px')};
	height: ${props => (props.height ? `${props.height}px` : '100px')};
	background: transparent;
	border-radius: 50%;
	&:before {
		content: '';
		display: block;
		width: ${props => (props.width ? `${props.width}px` : '100px')};
		height: ${props => (props.height ? `${props.height}px` : '100px')};
		background: ${props => (props.pColor ? props.pColor : 'red')};
		border-radius: 50%;
		animation: ${pulse} 3s linear infinite;
	}
	&:after {
		content: '';
		display: block;
		position: absolute;
		width: ${props => (props.width ? `${props.width}px` : '100px')};
		height: ${props => (props.height ? `${props.height}px` : '100px')};
		background: ${props => (props.sColor ? props.sColor : 'red')};
		border-radius: 50%;
		animation: ${pulse} 2.3s linear infinite;
	}
`;

PulseLoader.propTypes = {
  width: number,
  height: number,
  loading: bool,
  pColor: string,
  sColor: string
};

export default PulseLoader;
