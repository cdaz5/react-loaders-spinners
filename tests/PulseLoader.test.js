import React from 'react';
import renderer from 'react-test-renderer';

import PulseLoader from '../src/components/PulseLoader';

describe('PulseLoader', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<PulseLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
