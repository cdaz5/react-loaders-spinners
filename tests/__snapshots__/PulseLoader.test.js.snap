// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PulseLoader matches snapshot 1`] = `
<div
  className="sc-bdVaJa cNGzqq"
  height={100}
  width={100}
/>
`;
